packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml

```
@@ -72,14 +66,80 @@ processors:
+  - script:
+      lang: painless
+      tag: script-misp_decayed
+      if: ctx.misp?.attribute?.decay_score instanceof List && ctx.misp.attribute.decay_score.size() > 0
+      source: |
+        def decayed = 
+            ctx.misp.attribute.decay_score.stream()
+            .filter(t -> t.decayed == true)
+            .count()/1.0 >= (ctx.misp.attribute.decay_score.size())/2.0 ;
+        ctx.misp.attribute.decayed = decayed;
+        if (decayed == true) {
+            ctx.misp.attribute.decayed_at = ZonedDateTime.parse(ctx._tmp.event_ingested).minusMinutes(2L); 
+        }
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+# Script to set default expiration to delete lingering orphaned indicators in the destination indices.
+# Calculate max(misp.attribute.timestamp, misp.attribute.last_seen) as _tmp_max_time
+# Add default decayed_at to expire all IOCs after duration `ioc_expiration_duration` from `_tmp_max_time`. 
+# If user-provided value of `ioc_expiration_duration` is not in d, h, or m, default to 90d.
+# Set `decayed` as true if decayed_at is before ingest time i.e., already expired during ingestion.
+  - script:
+      lang: painless
+      tag: script-default-decayed_at
+      if: (ctx.misp?.attribute?.decayed_at == null && ctx._conf?.ioc_expiration_duration != null && ctx._conf.ioc_expiration_duration != '')
+      description: >
+        Add default decayed_at
+      source: >
+        def dur = ctx._conf.ioc_expiration_duration;
+        ZonedDateTime _tmp_decayed_at;
+        ZonedDateTime _tmp_timestamp = ZonedDateTime.ofInstant(Instant.ofEpochMilli(Long.parseLong(ctx.misp.attribute.timestamp) * 1000L), ZoneId.of('Z'));
+        ZonedDateTime _tmp_max_time = _tmp_timestamp;
+        if (ctx.misp.attribute.last_seen != null) {
+            ZonedDateTime _tmp_last_seen = ZonedDateTime.ofInstant(Instant.ofEpochMilli((ctx.misp.attribute.last_seen)/1000), ZoneId.of('Z'));
+            if (_tmp_max_time.isBefore(_tmp_last_seen)) {
+                _tmp_max_time = _tmp_last_seen;
+            }
+        }
+        if (dur instanceof String){
+          Character time_unit = dur.charAt(dur.length() - 1);
+          String time_value = dur.substring(0, dur.length());
+          if (time_unit == 'd') {
+            _tmp_decayed_at = _tmp_max_time.plusDays(Long.parseLong(time_value));
+          } else if (time_unit == 'h') {
+            _tmp_decayed_at = _tmp_max_time.plusHours(Long.parseLong(time_value));
+          } else if (time_unit == 'm') {
+            _tmp_decayed_at = _tmp_max_time.plusMinutes(Long.parseLong(time_value));
+          } else {
+            _tmp_decayed_at = _tmp_max_time.plusDays(90L);
```

Add an error to `error.message` here? If they accidentally type e.g. "120n" (for 120m) and we give them 90 days, that might be confusing.

---

The reason for this is to have atleast a default expiry of `90d` to avoid any indicators becoming orphaned and just be there indefinitely without expiry.

---

Yeah, I agree, but I think that a warning could be appended into the `errors.message` array at the same time.
```suggestion
            _tmp_decayed_at = _tmp_max_time.plusDays(90L);
            if (ctx.error == null) {
              ctx.error = new HashMap();
            }
            if (ctx.error.message == null) {
              ctx.error.message = new ArrayList();
            }
            ctx.error.message.add('invalid ioc_expiration_duration: using 90 days');
```

---

Thanks. That makes sense. Will do that 👍🏼

---

Thanks for checking this which lead to a bug. 
The character needs to be casted since `'d', 'm', 's'` are considered as string literals in painless.